### PR TITLE
implement throttle

### PR DESCRIPTION
**Pull request by piever**
Looking at [this](https://github.com/JuliaGizmos/Interact.jl/issues/225) I've noticed that we do not have `throttle` in Observables and I thought it belongs in this package (but am happy to keep it in InteractBase if we want to keep this package more minimalistic).

### CODE REVIEW

**Comment by shashi** [comment]
I think we can have it here

**Comment by korsbo** [comment]
I found my way to this PR because I needed a throttle function. This works like a charm. Cheers!

**Comment by piever** [comment]
I've changed the implementation to no longer require the `while` trick, so this should not create problems with garbage collection: as soon as the throttled is updated, a timer is started that prevents the update in the following time window.

File: test/runtests.jl [hunk 1]
```diff
@@ -43,3 +43,17 @@ end
...
+    obs = Observable(1)
+    throttled = throttle(10, obs)
+    @test throttled[] == 1
+    obs[] = 2
+    sleep(0.1)
+    @test throttled[] == 2
+    obs[] = 3
+    sleep(5)
+    @test throttled[] == 2
+    sleep(10)
```
Comment (shashi) [inline test/runtests.jl]: I think these numbers are too big :sweat_smile: But then I'm not the most patient person.

**Comment by shashi** [comment]
Another way to "throttle" (there's probably some other name for it) updates is to maintain a buffer of at most `n` items and keep in it only the last `n` updates.

```julia
using Observables

function latest(s, n=3)
    buffer = Any[]
    cond = Condition()
    on(s) do x
        if length(buffer) < n
            push!(buffer, x)
        else
            while length(buffer) >= n
                pop!(buffer)
            end
            unshift!(buffer, x)
        end
        notify(cond)
    end
    o = Observable{Any}(s[])
    @async while true
        while !isempty(buffer)
            upd = pop!(buffer)
            o[] = upd
        end
        wait(cond)
    end
    o
end

s = Observable{Any}(0)
o = latest(s)

on(o) do x
    sleep(1)
    println(x)
end

s[]=1
s[]=2
s[]=3
s[]=4
s[]=5
s[]=6
s[]=7
s[]=8
s[]=9
s[]=10
```
This will print: 1,8,9,10. The first update is propagated, then the next ones come way too quickly so the buffer gets truncated to the last 3. Notice also that `s` updates asynchronously.

File: test/runtests.jl (hunk 1 above)
Comment (piever) [reply test/runtests.jl]: lol, I sped  it up by a factor of five.

**Comment by piever** [comment]
That's interesting: I like `async_latest` from the other PR, but I agree that they are different concepts. We should probably keep both versions with different names?

**Comment by shashi** [comment]
Yeah the two are different things so we should keep them both!

**Comment by shashi** [comment]
Go ahead and merge this if you don't intend any more changes.

**Pull request merged**